File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/AzureActiveDirectoryB2C.NET8.0/Initialization/Starter.fs
namespace Initialization

open AuthZI.MicrosoftEntra
open AuthZI.Deploy.MicrosoftEntra.Configuration
open AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common
open RootConfiguration
open System.Threading.Tasks

[<assembly: Orleans.ApplicationPartAttribute("AuthZI.Tests.MicrosoftOrleans.Grains")>]
()

type TestDataPipelineStartup() =
    interface Xunit.v3.ITestPipelineStartup with
        member _.StartAsync(_diagnosticMessageSink: Xunit.Sdk.IMessageSink) =
            let credentialsJson =
                TestDataInitialization.getCredentialsJson "microsoftAzureADB2CCredentials" Literals.azureADB2C1Json

            TestDataInitialization.initializeUsersOnly credentialsJson
            ValueTask()

        member _.StopAsync() = ValueTask()

type Starter() =
    do ()

[<assembly: Xunit.v3.TestPipelineStartup(typeof<TestDataPipelineStartup>)>]
[<assembly: Xunit.AssemblyFixture(typeof<Starter>)>]
()

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.CSharp/Initialization/Starter.cs
using AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common.Initialization;
using Orleans;
using Xunit;

[assembly: ApplicationPart("AuthZI.Tests.MicrosoftOrleans.Grains")]
[assembly: AssemblyFixture(typeof(MicrosoftEntraIdTestFixture))]
[assembly: CollectionBehavior(DisableTestParallelization = true)]
